lib/db.js

```diff
@@ -239,39 +246,77 @@ class LimitDBRedis extends EventEmitter {
           count *= (bucketKeyConfig.skip_n_calls + 1);
         }
       }
-
-
     }
 
-    this.redis.take(key,
-      bucketKeyConfig.ms_per_interval || 0,
-      bucketKeyConfig.size,
-      count,
-      Math.ceil(bucketKeyConfig.ttl || this.globalTTL),
-      bucketKeyConfig.drip_interval || 0,
-      (err, results) => {
-        if (err) {
-          return callback(err);
-        }
-
-        const remaining = parseInt(results[0], 10);
-        const conformant = parseInt(results[1], 10) ? true : false;
-        const currentMS = parseInt(results[2], 10);
-        const reset = parseInt(results[3], 10);
-        const res = {
-          conformant,
-          remaining,
-          reset: Math.ceil(reset / 1000),
-          limit: bucketKeyConfig.size,
-          delayed: false,
-        };
-
-        if (bucketKeyConfig.skip_n_calls > 0) {
-          this.callCounts.set(key, { res, count: 0 });
-        }
+    if (bucket.elevated_limits) {
+      if (!erlIsActiveKey)  {
+        return callback(new Error('erlIsActiveKey is required for elevated limits'), null);
+      }
+      this.redis.takeElevated(key, erlIsActiveKey,
+          bucketKeyConfig.ms_per_interval || 0,
+          bucketKeyConfig.size,
+          count,
+          Math.ceil(bucketKeyConfig.ttl || this.globalTTL),
+          bucketKeyConfig.drip_interval || 0,
+          bucket.elevated_limits.ms_per_interval || 0,
+          bucket.elevated_limits.size,
+          bucketKeyConfig.elevated_limits.activation_period_minutes || 15,
```

Comment:
we can use a config variable for default `activation_period_minutes`

Reply:
Do you mean to set the default value while resolving the bucketKeyConfig itself?